### PR TITLE
Add explicit k8s namespace for creating the ARC GitHub API PAT secret

### LINKED ISSUE #29217: Add explicit k8s namespace for creating the ARC GitHub API PAT secret
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api

### What part(s) of the article would you like to see updated?

kubectl examples

### Additional information

The examples use a generic "my_namespace" and this is confusing.

The quick start is using 2 namespaces:

* arc-systems - control
* arc-runners - workers

I think it helps to have explicit info about where to create the secret

### PR DESCRIPTION
**Pull request by adiroiban**
### Why:

Closes: #29217

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Add explicit namespace in example.

### Todo

- [ ] Also update the example when using a GitHub Action ... a need to check the source of that content.

### Check off the following:

- [ ] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by mchammer01** [comment]
The update looks good to me. Checking with the relevant team as I think that we may need to make a couple more similar updates. Will report back once I know more 😃

**Comment by adiroiban** [comment]
Yes. This needs changes in other parts. I am not familiar with the build system for this site, so I just started this PR to start a conversation.

Maybe my suggestion is not correct.

But I think that is important to specificy in which namespace we need to create the secret.

My understanding, is that with the new ARC controller, the secret is only needed in the controller.

The runners will receive an already computed token needed to join the list of available runners.

**Comment by mchammer01** [comment]
@adiroiban - thanks for starting the conversation 😃 
Someone from the relevant focus team, who knows more than me on GitHub Actions, will review this as soon as they can.

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ ARC can use {% data variables.product.pat_v1_plural %} to register self-hosted r
 
    ```bash copy
    kubectl create secret generic pre-defined-secret \
-      --namespace=my_namespace \
+      --namespace=arc-systems \
```
Comment (Link-) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md:72]:
```suggestion
      --namespace=arc-runners \
```

**Comment by Link-** [comment]
Thank you @adiroiban for your contribution to making our docs better!

**Review by Link- — CHANGES_REQUESTED** [review]
Before we merge this, it'll be awesome to also add note here to clarify why we used the `arc-runners` namespace as opposed to `arc-systems`

cs_2023-10-20_10-29-00

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ ARC can use {% data variables.product.pat_v1_plural %} to register self-hosted r
 
    ```bash copy
```
Comment (jc-clark) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md:70]:
```suggestion
   {% note %}
   
   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller.md#configuring-a-runner-scale-set
).".
   
   {% endnote %}
   
   ```bash copy
```

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md [hunk 3]
```diff
@@ -67,9 +67,15 @@ ARC can use {% data variables.product.pat_v1_plural %} to register self-hosted r
     {% endif %}
 1. To create a Kubernetes secret with the value of your {% data variables.product.pat_v1 %}, use the following command.
 
+   {% note %}
+   
+   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller.md#configuring-a-runner-scale-set)."
+   
+   {% endnote %}
+
```
Comment (jc-clark) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md:75]:
```suggestion
1. To create a Kubernetes secret with the value of your {% data variables.product.pat_v1 %}, use the following command.
   {% note %}
   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller.md#configuring-a-runner-scale-set)."
   {% endnote %}
```

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md [hunk 4]
```diff
@@ -67,9 +67,15 @@ ARC can use {% data variables.product.pat_v1_plural %} to register self-hosted r
     {% endif %}
 1. To create a Kubernetes secret with the value of your {% data variables.product.pat_v1 %}, use the following command.
 
+   {% note %}
+   
+   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller.md#configuring-a-runner-scale-set)."
```
Comment (jc-clark) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md:72]:
```suggestion
   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#configuring-a-runner-scale-set)."
```

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md [hunk 5]
```diff
@@ -67,9 +67,15 @@ ARC can use {% data variables.product.pat_v1_plural %} to register self-hosted r
     {% endif %}
 1. To create a Kubernetes secret with the value of your {% data variables.product.pat_v1 %}, use the following command.
 
+   {% note %}
+   
+   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#configuring-a-runner-scale-set)."
+   
+   {% endnote %}
+
```
Comment (jc-clark) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md:75]:
```suggestion
   {% note %}
   
   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#configuring-a-runner-scale-set)."

   {% endnote %}
   
```

File: content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md [hunk 6]
```diff
@@ -67,9 +67,15 @@ ARC can use {% data variables.product.pat_v1_plural %} to register self-hosted r
     {% endif %}
 1. To create a Kubernetes secret with the value of your {% data variables.product.pat_v1 %}, use the following command.
 
+   {% note %}
+   
+   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#configuring-a-runner-scale-set)."
+
+   {% endnote %}
+
```
Comment (jc-clark) [inline content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/authenticating-to-the-github-api.md:75]:
```suggestion
   {% note %}

   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#configuring-a-runner-scale-set)."

   {% endnote %}

```

**Comment by jc-clark** [comment]
@Link- Thank you for the review! Let us know what you think of the changes from the latest commits. If it looks good to you, I can help merge this content 👍

**Comment by adiroiban** [comment]
Also, the "my_namespace" is refered in 2 places .

In the diff , it is only updated once.

Maybe that is ok.

image


image

**Comment by adiroiban** [comment]
I have tested it and is working ok.


Thanks for the fix :)

-------

I have previously reported an error, but it was something else.

It's a bit hard to troubleshoot errors that happen at pod `run` command level.
If a pod is scheduled by the controller but it fails to start, is not easy to track with with a barebone k8s cluster, as you can't retrie the logs post-mortem. 

ARC works. Thanks. Much appreciated

My issue was with the image `run` command. Now solved.

**Comment by Link-** [comment]
Good point @adiroiban - @jc-clark I'm sorry to drag this a bit further, the current changes are 👍 can we please also introduce these changes?

### Change 1

cs_2023-10-21_13-47-30

### Change 2

cs_2023-10-21_13-47-48

### Change 3

cs_2023-10-21_13-48-11

**Comment by jc-clark** [comment]
👋 Hello @adiroiban! I don't have permissions to commit changes to your pull request directly. Can you help us make the changes outlined by @Link- [above](https://github.com/github/docs/pull/29218#issuecomment-1773767052)? 

We will need to replace instances of `my_namespace` with `arc-runners` in those three locations. 

Let me know if you run into any issues adding this change to your pull request. And thank you in advance!

**Comment by adiroiban** [comment]
Hi.

Sorry for the delay... this repo is huge :)

I got it on my computer and and I hope I got it right.

```
$ grep my_namespace  content/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/*
```

**Comment by jc-clark** [comment]
No problem @adiroiban and apologies, I was not very clear in my last comment, so we missed part of @Link-'s suggestion. Sorry about that! 

I think we need to add the note about creating the secret in the right namespace to all three locations where we use this `arc-runners` example. 

To show the same information in multiple locations, we use [reusables](https://docs.github.com/en/contributing/writing-for-github-docs/creating-reusable-content). 

These are the steps to create a new reusable file and then reference it in multiple locations: 

1. Copy the following information into a new `.md` file in the [`data/reusables/actions` directory](https://github.com/github/docs/tree/main/data/reusables/actions directory). Save the file and name it `arc-runners-secret-namespace.md`. 

   ```
   {% note %}

   **Note:** Create the secret in the same namespace where the `gha-runner-scale-set` chart is installed. In this example, the namespace is `arc-runners` to match the quickstart documentation. For more information, see "[AUTOTITLE](/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller#configuring-a-runner-scale-set)."

   {% endnote %}
   ```
1. Copy `{% data reusables.actions.arc-runners-secret-namespace %}` to your clipboard and paste it above the three locations where we use `arc-runners`
1. Commit your changes to this PR. 

Let me know if you run into any issues with this, I'll be happy to help out. And thank you for your patience while I missed this note before 😅

**Comment by jc-clark** [comment]
Great catch @adiroiban! I think that name you suggested for the reusable would work perfectly.

**Comment by adiroiban** [comment]
Sorry. I didn't had much time for this. Still spending the whole week getting ARC un and running on my homelab.

I have pushed a change. I hope it's good enough to merge.

**Comment by jc-clark** [comment]
This looks amazing @adiroiban! Thanks for the changes. 🪩

We currently have a repo freeze happening, so I'll be able to merge this next week, when we are clear to publish docs again.

**Review by jc-clark — APPROVED** [review]

**Pull request merged**